recipes/libelfin/all/conanfile.py

```diff
@@ -0,0 +1,57 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class LibelfinConan(ConanFile):
+    name = "libelfin"
+    description = "C++11 library for reading ELF binaries and DWARFv4 debug information"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aclements/libelfin"
+    license = "MIT"
+    topics = ("conan", "elf", "dwarf", "libelfin")
+
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    exports_sources = "CMakeLists.txt", "patches/*"
+
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        if self.should_build:
+            cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
```suggestion
        self.cpp_info.libs = ["elf++", "dwarf++"]
```

Reply:
Why not to use `collect_libs`? With collect_libs we don't have to duplicate the list of installed libraries, it all comes from CMakeLists.txt.

Reply:
I think in this case it's ok, but when libraries are dependent on this `tools.collect_libs` might put them in the wrong order.
And order matters when linking to static libraries.

Reply:
This makes perfect sense. We have `dwarf++` depends on `elf++`, so then must go in-order. Else basic `cmake` generator` can fail if ordering change. This shouldn't affect if we use `CONAN_PKG::dwarf++`, which still requires explicit dependency. Need better component management to make this right.